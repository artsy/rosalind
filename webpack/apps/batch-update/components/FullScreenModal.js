import React from 'react'
import styled from 'styled-components'
import { ESC } from 'lib/keycodes.js'

const Modal = styled.div`
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;
  background: white;

  // modal is closed
  visibility: hidden;
  opacity: 0;
  transform: scale(0.95);
  transition: all 0.25s;

  // modal is open
  &.modal-open {
    visibility: visible;
    opacity: 1;
    transform: scale(1);
    transition: all 0.125s;
  }

`
Modal.displayName = 'Modal'

class FullScreenModal extends React.Component {
  constructor (props) {
    super(props)
    this.handleKeyup = this.handleKeyup.bind(this)
  }

  componentWillUpdate ({isOpen: willBeOpen}) {
    const { isOpen } = this.props
    if (!isOpen && willBeOpen) {
      // modal is opening
      window.addEventListener('keyup', this.handleKeyup)
    }
    if (isOpen && !willBeOpen) {
      // modal is closing
      window.removeEventListener('keyup', this.handleKeyup)
    }
  }

  handleKeyup (e) {
    if (e.keyCode === ESC) {
      this.props.onDismiss()
    }
  }

  render () {
    const { isOpen, children } = this.props

    const className = isOpen ? 'modal-open' : null
    return (
      <Modal className={className}>
        {children}
      </Modal>
    )
  }
}

export default FullScreenModal
