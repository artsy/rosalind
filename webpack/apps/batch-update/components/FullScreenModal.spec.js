import React from 'react'
import renderer from 'react-test-renderer'
import FullScreenModal from './FullScreenModal'

it('renders correctly when closed', () => {
  const rendered = renderer.create(
    <FullScreenModal>
      I am closed
    </FullScreenModal>
  )
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when open', () => {
  const rendered = renderer.create(
    <FullScreenModal isOpen>
      I am open
    </FullScreenModal>
  )
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
