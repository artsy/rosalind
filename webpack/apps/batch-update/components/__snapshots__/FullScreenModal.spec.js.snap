// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when closed 1`] = `
.c0 {
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;
  background: white;
  overflow: scroll;
  visibility: hidden;
  opacity: 0;
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.c0.modal-open {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 0.125s;
  transition: all 0.125s;
}

<div
  className="c0"
>
  I am closed
</div>
`;

exports[`renders correctly when open 1`] = `
.c0 {
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;
  background: white;
  overflow: scroll;
  visibility: hidden;
  opacity: 0;
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.c0.modal-open {
  visibility: visible;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: all 0.125s;
  transition: all 0.125s;
}

<div
  className="modal-open c0"
>
  I am open
</div>
`;
